src/common.rs

```diff
@@ -57,7 +61,7 @@ impl FileEntry {
 #[derive(Clone, Debug, PartialEq)]
 pub struct LineEntry {
     pub number: usize,
-    pub file_path: PathBuf,
+    pub file: FileEntry,
```

Comment:
It would be great, but right now I don't know how to implement it with less code.
Maybe I will do it later or you may help me with it 😉
What do you think about it?

Reply:
> It would be great, but right now I don't know how to implement it with less code.
> Maybe I will do it later or you may help me with it wink
> What do you think about it?

Manually setting the lifetime really seems rather complicated in this case, Rc seems to be a faster option, but there are still a lot of changes in the code ...

I tried to [implement everything on Rc and it's really big changes](https://github.com/dotenv-linter/dotenv-linter/compare/refactoring...DDtKey:refactoring), but fortunately, all the tests pass, but I do not risk doing a pull request out of expediency :scream: 

I propose to accept your option and possibly revise it in future iterations.